### PR TITLE
feat(update): add InstanceChargeType support SPOTPAID

### PR DESCRIPTION
**Pull request by ysicing**
build instance charge  support `SPOTPAID` type

Charge type of cvm, values can be `POSTPAID_BY_HOUR` (default)  `SPOTPAID`

Signed-off-by: ysicing <i@ysicing.me>

### CODE REVIEW

File: builder/tencentcloud/cvm/step_run_instance.go [hunk 1]
```diff
@@ -51,15 +52,18 @@ func (s *stepRunInstance) Run(ctx context.Context, state multistep.StateBag) mul
...
 
 	// config RunInstances parameters
-	POSTPAID_BY_HOUR := "POSTPAID_BY_HOUR"
 	req := cvm.NewRunInstancesRequest()
 	if s.ZoneId != "" {
 		req.Placement = &cvm.Placement{
 			Zone: &s.ZoneId,
 		}
 	}
+	instance_charge_type := s.InstanceChargeType
```
Comment (lbajolet-hashicorp) [inline builder/tencentcloud/cvm/step_run_instance.go:61]: For local variables, Go tends to encourage camelCase in terms of naming, could you rename it to `instanceChargeType`?

**Review by lbajolet-hashicorp — COMMENTED** [review]
Hey @ysicing,

Thanks for the contribution, I just left a couple comments, but outside from that, the code looks good to me.

When they're addressed, let me know, I'll do another pass, and if the code looks good, we can merge this.

File: builder/tencentcloud/cvm/step_run_instance.go [hunk 2]
```diff
@@ -51,15 +52,18 @@ func (s *stepRunInstance) Run(ctx context.Context, state multistep.StateBag) mul
...
+	if instance_charge_type != "SPOTPAID" {
```
[the remark this answers is not in the document]
Comment (ysicing) [reply builder/tencentcloud/cvm/step_run_instance.go]: You're right

File: builder/tencentcloud/cvm/step_run_instance.go (hunk 1 above)
Comment (ysicing) [reply builder/tencentcloud/cvm/step_run_instance.go]: fixed

**Review by lbajolet-hashicorp — APPROVED** [review]
Hi @ysicing,

First off apologies for just coming back to this PR, I forgot about it, let's see what we can do to get it merged.

We'll be releasing a new version of the plugin, given that your code is almost ready to merge, I'd very much like to get that in before committing the release.

From the looks of it, you included the `.docs` directory in your changes, which are not supposed to be tracked. This is produced when running `make generate`, and should not be included in the changes here.

I'll see to add an exception to the gitignore for them, in the meantime, can I ask you to remove this from the PR?

Thanks, and apologies again for the delay.

**Review by lbajolet-hashicorp — APPROVED** [review]
LGTM, thanks for the reroll @ysicing, appreciated.

I'll merge this and release the new version of the plugin after

**Pull request merged**